addons/barcodes_gs1_nomenclature/models/barcode_nomenclature.py

```diff
@@ -124,3 +124,41 @@ def parse_barcode(self, barcode):
         if self.is_gs1_nomenclature:
             return self.gs1_decompose_extanded(barcode)
         return super().parse_barcode(barcode)
+
+    @api.model
+    def _preprocess_gs1_search_args(self, args, barcode_types, field='barcode'):
```

Comment:
Instead of silently changing `args`, it should be better to return it.
It won't make any changes but it will be clearer when reading the code than `args` can be modified after calling `_preprocess_gs1_search_args`